pilot/pkg/serviceregistry/kube/controller/controller.go

```diff
@@ -493,6 +494,7 @@ func (c *Controller) InstancesByPort(svc *model.Service, reqSvcPort int,
 						Service:        svc,
 						Labels:         podLabels,
 						ServiceAccount: sa,
+						MTLSReady:      mtlsReady,
```

Comment:
you have to add this to Endpoint also. Else no point in sticking the mtlsReady in the endpoint struct. or you can remove it from the endpoint